app/scripts/lib/security-provider-helpers.ts

```diff
@@ -1,35 +1,46 @@
+import { Json } from '@metamask/utils';
+import { MessageParams } from '@metamask/message-manager';
 import getFetchWithTimeout from '../../../shared/modules/fetch-with-timeout';
 import { MESSAGE_TYPE } from '../../../shared/constants/app';
 
 const fetchWithTimeout = getFetchWithTimeout();
 
+export interface RequestData {
+  origin: string;
+  msgParams?: MessageParams;
```

Comment:
If we always need at least one of the message params properties, could we instead use a type with a union to ensure one is always provided?

Reply:
I've included now `MessageRequestData` and `MessageTransactionData`  types however I had to set the properties that should not be present in the type as never so it does not need to check the type again further on.